### ISSUE TITLE
3.1.1 Exception loading animated gif

### ISSUE THREAD

**Issue by crumpf** [facebook/fresco#2747]
### Description

Updating fresco from `3.1.0` to `3.1.1` throws `java.lang.UnsatisfiedLinkError` exception when trying to load an animated gif. Static images seem fine, though have not exhaustively tested different file formats.

```
FATAL EXCEPTION: FrescoDecodeExecutor-2
                                                                                                    Process: com.example.app, PID: 22637
                                                                                                    java.lang.UnsatisfiedLinkError: dlopen failed: library "libgifimage.so" not found
                                                                                                    	at java.lang.Runtime.loadLibrary0(Runtime.java:1077)
                                                                                                    	at java.lang.Runtime.loadLibrary0(Runtime.java:998)
                                                                                                    	at java.lang.System.loadLibrary(System.java:1656)
                                                                                                    	at com.facebook.soloader.nativeloader.SystemDelegate.loadLibrary(SystemDelegate.java:24)
                                                                                                    	at com.facebook.soloader.nativeloader.NativeLoader.loadLibrary(NativeLoader.java:52)
                                                                                                    	at com.facebook.soloader.nativeloader.NativeLoader.loadLibrary(NativeLoader.java:30)
                                                                                                    	at com.facebook.animated.gif.GifImage.ensure(GifImage.java:48)
                                                                                                    	at com.facebook.animated.gif.GifImage.createFromNativeMemory(GifImage.java:97)
                                                                                                    	at com.facebook.animated.gif.GifImage.decodeFromNativeMemory(GifImage.java:122)
                                                                                                    	at com.facebook.imagepipeline.animated.factory.AnimatedImageFactoryImpl.decodeGif(AnimatedImageFactoryImpl.java:102)
                                                                                                    	at com.facebook.fresco.animation.factory.AnimatedFactoryV2Impl$1.decode(AnimatedFactoryV2Impl.java:108)
                                                                                                    	at com.facebook.imagepipeline.decoder.DefaultImageDecoder.decodeGif(DefaultImageDecoder.java:171)
                                                                                                    	at com.facebook.imagepipeline.decoder.DefaultImageDecoder$1.decode(DefaultImageDecoder.java:76)
                                                                                                    	at com.facebook.imagepipeline.decoder.DefaultImageDecoder.decode(DefaultImageDecoder.java:152)
                                                                                                    	at com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder.internalDecode(DecodeProducer.kt:260)
                                                                                                    	at com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder.doDecode(DecodeProducer.kt:202)
                                                                                                    	at com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder._init_$lambda$2(DecodeProducer.kt:414)
                                                                                                    	at com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder.$r8$lambda$KZ-YIhcmQg_RJ489kdh-N9D6sow(Unknown Source:0)
                                                                                                    	at com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder$$ExternalSyntheticLambda0.run(Unknown Source:6)
                                                                                                    	at com.facebook.imagepipeline.producers.JobScheduler.doJob(JobScheduler.java:231)
                                                                                                    	at com.facebook.imagepipeline.producers.JobScheduler.access$000(JobScheduler.java:28)
                                                                                                    	at com.facebook.imagepipeline.producers.JobScheduler$1.run(JobScheduler.java:97)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
                                                                                                    	at com.facebook.imagepipeline.core.PriorityThreadFactory.newThread$lambda$0(PriorityThreadFactory.kt:37)
                                                                                                    	at com.facebook.imagepipeline.core.PriorityThreadFactory.$r8$lambda$IPp7Vm9a1KDy8D4770JTjI9qOG4(Unknown Source:0)
                                                                                                    	at com.facebook.imagepipeline.core.PriorityThreadFactory$$ExternalSyntheticLambda0.run(Unknown Source:4)
                                                                                                    	at java.lang.Thread.run(Thread.java:920)
```                                                                                                    	

### Reproduction

Gradle dependencies:
```
    implementation "com.facebook.fresco:fresco:3.1.1"
    implementation "com.facebook.fresco:animated-gif:3.1.1"
    implementation "com.facebook.fresco:animated-webp:3.1.1"
    implementation "com.facebook.fresco:webpsupport:3.1.1"
    implementation "com.facebook.fresco:middleware:3.1.1" // https://github.com/facebook/fresco/issues/2714#issuecomment-1476700754
    // https://github.com/facebook/fresco/issues/2722 seems like libnative-imagetranscoder.so was omitted from Fresco 3.1.1 so use 3.1.0 until that's fixed
    implementation('com.facebook.fresco:nativeimagetranscoder') {
        version {
            strictly '3.1.0'
        }
    }
```

URL of gif to reproduce: https://c.tenor.com/CLcaCnXxUW8AAAAC/tenor.gif

Usage similar to:

```
val draweeController = Fresco.newDraweeControllerBuilder()
    .setUri(uri)
    .setAutoPlayAnimations(true)
    .build()
myDraweeHolder.controller = draweeController
```

### Solution

🤷‍♂️

### Additional Information

* Fresco version: 3.1.1
* Platform version: Pixel 6 API 32

**Comment by crumpf** [comment]
Futzed around with this a bit more. Set `com.facebook.fresco:animated-gif` back to `3.1.0` to avoid the`java.lang.UnsatisfiedLinkError: dlopen failed: library "libgifimage.so" not found` exception.

e.g.:

```
    implementation "com.facebook.fresco:fresco:3.1.1"
    implementation "com.facebook.fresco:animated-gif:3.1.0" // 3.1.0 because 3.1.1 has  `libgifimage.so not found` exception 
    implementation "com.facebook.fresco:animated-webp:3.1.1"
    implementation "com.facebook.fresco:webpsupport:3.1.1"
    implementation "com.facebook.fresco:middleware:3.1.1" // https://github.com/facebook/fresco/issues/2714#issuecomment-1476700754
    // https://github.com/facebook/fresco/issues/2722 seems like libnative-imagetranscoder.so was omitted from Fresco 3.1.1 so use 3.1.0 until that's fixed
    implementation('com.facebook.fresco:nativeimagetranscoder') {
        version {
            strictly '3.1.0'
        }
    }
```

**Comment by crumpf** [comment]
This does not appear to be fixed in `3.1.2`
